client/src/containers/ArtistDetail.js

```diff
@@ -51,25 +53,78 @@ const ArtistContent = styled.div`
 `;
 
 class ArtistDetail extends Component {
-  render() {
+  componentDidMount = () => {
+    const { fetchArtist, artistId } = this.props;
+
+    fetchArtist(artistId);
+  };
+
+  onClickSetPlaylist = songIndex => {
+    const { setArtistSongsPlaylist, music } = this.props;
+    const artistSongs = music.artistInfo.songs;
+
+    setArtistSongsPlaylist(artistSongs, songIndex);
+  };
+
+  render = () => {
+    const {
+      music: { artistInfo },
+      player,
+      onClickPlayTrack,
+      onClickPauseTrack,
+      isPlaying,
+      onClickSetArtistAlbumSongs,
+      onClickPauseArtistAlbumSongs
+    } = this.props;
+
+    if (!artistInfo.songs) {
+      return null;
+    }
+
     return (
       <>
-        <ArtistHeader>
-          <ArtistListenersCount>273,591 MONTHLY LISTENERS</ArtistListenersCount>
-          <ArtistName>Kyog, Lot</ArtistName>
+        <ArtistHeader src={artistInfo.artist.artistCover}>
+          <ArtistListenersCount>
+            {artistInfo.artist_plays || 0} PLAYS
+          </ArtistListenersCount>
+          <ArtistName>{artistInfo.artist.name}</ArtistName>
           <Button half color="white" bgColor="green" borderColor="green">
             PLAY
           </Button>
         </ArtistHeader>
         <ArtistContent>
           <SectionTitle>Popular</SectionTitle>
-          <ArtistTracks />
+          <ArtistTracks
+            songs={artistInfo.songs}
+            onClickSetPlaylist={this.onClickSetPlaylist}
+            onClickPlayTrack={onClickPlayTrack}
+            onClickPauseTrack={onClickPauseTrack}
+            currentIndex={player.currentPlaylist[player.currentIndex].id}
+            isPlaying={isPlaying}
+          />
           <SectionTitle>Albums</SectionTitle>
-          <ArtistAlbums />
+          <ArtistAlbums
+            isPlaying={player.isPlaying}
+            currentlyPlayingAlbumId={player.currentlyPlayingAlbumId}
+            albums={artistInfo.albums}
+            onClickSetArtistAlbumSongs={onClickSetArtistAlbumSongs}
+            onClickPauseArtistAlbumSongs={onClickPauseArtistAlbumSongs}
+          />
         </ArtistContent>
       </>
     );
-  }
+  };
 }
 
-export default ArtistDetail;
+const mapStateToProps = ({ player, music }) => ({ player, music });
```

Comment:
its not good practice to import whole reducer if its not really needed. with this approach, if any prop inside this reducer is updated, this component will re-render which is not what we want for sure, so be careful when importing like this. Make sure you need all props from the reducer.